tauri-app/src/lib/components/ModalVaultDeposit.svelte

```diff
@@ -3,91 +3,139 @@
+  import ModalExecute from './ModalExecute.svelte';
 
   export let open = false;
   export let vault: TokenVaultDetail | TokenVaultListItem;
   let amount: bigint;
   let isSubmitting = false;
+  let selectWallet = false;
 
   function reset() {
-    amount = 0n;
-    isSubmitting = false;
     open = false;
+    if (!isSubmitting) {
+      amount = 0n;
+      selectWallet = false;
+    }
   }
 
-  async function execute() {
+  async function executeLedger() {
     isSubmitting = true;
     try {
       await vaultDeposit(vault.vault_id, vault.token.id, amount);
-      reset();
       // eslint-disable-next-line no-empty
     } catch (e) {}
     isSubmitting = false;
+    reset();
   }
-</script>
 
-<Modal title="Deposit to Vault" bind:open outsideclose size="sm" on:close={reset}>
-  <div>
-    <h5 class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white">
-      Vault ID
-    </h5>
-    <p class="break-all font-normal leading-tight text-gray-700 dark:text-gray-400">
-      {bigintStringToHex(vault.vault_id)}
-    </p>
-  </div>
+  async function executeWalletconnect() {
+    isSubmitting = true;
+    try {
+      if (!$orderbookAddress) throw Error("Select an orderbook to deposit");
+      const allowance = await checkAllowance(amount, vault.token.id, $orderbookAddress);
+      if (!allowance) {
+        const approveCalldata = await vaultDepositApproveCalldata(vault.vault_id, vault.token.id, amount) as Uint8Array;
+        const approveTx = await ethersExecute(approveCalldata, vault.token.id);
+        toasts.success("Approve Transaction sent successfully!");
+        await approveTx.wait(1);
+      }
 
-  <div>
-    <h5 class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white">
-      Token
-    </h5>
-    <p class="break-all font-normal leading-tight text-gray-700 dark:text-gray-400">
-      {vault.token.name}
-    </p>
-  </div>
+      const depositCalldata = await vaultDepositCalldata(vault.vault_id, vault.token.id, amount) as Uint8Array;
+      const depositTx = await ethersExecute(depositCalldata, $orderbookAddress);
+      toasts.success("Transaction sent successfully!");
+      await depositTx.wait(1);
 
-  <div>
-    <h5 class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white">
-      Owner
-    </h5>
-    <p class="break-all font-normal leading-tight text-gray-700 dark:text-gray-400">
-      {vault.owner.id}
-    </p>
-  </div>
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.log(e);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      if (typeof e === "object" && (e as any)?.reason) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        toasts.error(`Transaction failed, reason: ${(e as any).reason}`);
+      }
+      else if (typeof e === "string") toasts.error(e);
+      else if (e instanceof Error) toasts.error(e.message);
+      else toasts.error("Transaction failed!");
```

Comment:
If the logic to handle and error and convert it into a string is shared between multiple components, can we pull it out into a helper util?

Reply:
i implemented this